Engine/ac/display.cpp

```diff
@@ -446,39 +445,112 @@ bool ShouldAntiAliasText() {
     return (game.options[OPT_ANTIALIASFONTS] != 0);
 }
 
-void wouttext_outline(Common::Bitmap *ds, int xxp, int yyp, int usingfont, color_t text_color, const char *texx) {
-    
-    color_t outline_color = ds->GetCompatibleColor(play.speech_text_shadow);
-    if (get_font_outline(usingfont) >= 0) {
-        // MACPORT FIX 9/6/5: cast
-        wouttextxy(ds, xxp, yyp, (int)get_font_outline(usingfont), outline_color, texx);
+// Make semi-transparent bits much more opaque
+void wouttextxy_AutoOutline_Semitransparent2Opaque(AGS::Common::Bitmap *map)
+{
+    if (map->GetColorDepth() < 32)
+        return; // such maps don't feature partial transparency
+    size_t const width = map->GetWidth();
+    size_t const height = map->GetHeight();
+
+    for (size_t y = 0; y < height; y++)
+    {
+        int32 *sc_line = reinterpret_cast<int32 *>(map->GetScanLineForWriting(y));
+        for (size_t x = 0; x < width; x++)
+        {
+            int32 &px = sc_line[x];
+            int const transparency = geta(px);
+            if (0 < transparency && transparency < 255)
+                px = makeacol32(
+                    getr32(px), 
+                    getg32(px), 
+                    getb32(px), 
+                    std::min(85 + transparency * 2, 255));
+        }
     }
-    else if (get_font_outline(usingfont) == FONT_OUTLINE_AUTO) {
-        int outlineDist = 1;
+}
 
-        if (is_bitmap_font(usingfont) && get_font_scaling_mul(usingfont) > 1) {
-            // if it's a scaled up bitmap font, move the outline out more
-            outlineDist = get_fixed_pixel_size(1);
+// Draw outline that is calculated from the text font, not derived from an outline font
+void wouttextxy_AutoOutline(AGS::Common::Bitmap *ds, size_t font, int32_t color, const char *texx, int &xxp, int &yyp)
+{
+    int thickness = game.fonts.at(font).AutoOutlineThickness;
+    auto style = game.fonts.at(font).AutoOutlineStyle;
+    if (0 == thickness)
+    {
+        // Thickness that corresponds to 1 game pixel
+        thickness =
+            (is_bitmap_font(font) && get_font_scaling_mul(font) > 1) ?
+            get_fixed_pixel_size(1) : 1;
+        style = FontInfo::kSquared;
+    }
+    if (thickness <= 0)
+        return; 
+
+    size_t const t_width = wgettextwidth(texx, font);
+    size_t const t_height = wgettextheight(texx, font);
+    AGS::Common::Bitmap *outline_stencil =
+        AGS::Common::BitmapHelper::CreateTransparentBitmap(t_width, t_height + 2 * thickness, ds->GetColorDepth());
```

Comment:
There's no need to use these long notations like AGS::Common::BitmapHelper, and similar. Instead you could add
`using namespace AGS::Common;` at the beginning of the cpp file and omit that everywhere.